uta_tools/uta_tools.py

```diff
@@ -42,3 +45,392 @@ def __init__(self, seqrepo_data_path: str = SEQREPO_DATA_PATH,
+            return None
+        (tx_exons, exon_start, exon_end) = tx_exon_start_end
+
+        tx_exon_coords = self.uta_db.get_tx_exon_coords(
+            tx_exons, exon_start, exon_end)
+        if not tx_exon_coords:
+            return None
+        tx_exon_start, tx_exon_end = tx_exon_coords
+
+        alt_ac_start_end = await self.uta_db.get_alt_ac_start_and_end(
+            transcript, tx_exon_start, tx_exon_end, gene=gene)
+        if not alt_ac_start_end:
+            return None
+        alt_ac_start, alt_ac_end = alt_ac_start_end
+
+        start = alt_ac_start[3]
+        end = alt_ac_end[2]
+        strand = alt_ac_start[4]
+        if strand == -1:
+            start_offset = exon_start_offset * -1
+            end_offset = exon_end_offset * -1
+        else:
+            start_offset = exon_start_offset
+            end_offset = exon_end_offset
+        start += start_offset
+        end += end_offset
+
+        gene = alt_ac_start[0]
+        chromosome = alt_ac_start[1]
+        if gene is None or chromosome is None:
+            return None
+        return GenomicData(
+            gene=gene,
+            chr=chromosome,
+            start=start,
+            end=end,
+            exon_start=exon_start,
+            exon_start_offset=exon_start_offset,
+            exon_end=exon_end,
+            exon_end_offset=exon_end_offset,
+            transcript=transcript
+        )
+
+    async def genomic_to_transcript(self, chromosome: Union[str, int],
+                                    start: int, end: int,
+                                    strand: int = None, transcript: str = None,
+                                    gene: str = None,
+                                    residue_mode: ResidueMode = ResidueMode.RESIDUE,  # noqa: E501
+                                    *args, **kwargs) -> Optional[GenomicData]:
+        """Get transcript data for genomic data.
+        MANE Transcript data will be returned iff `transcript` is not supplied.
+            `gene` must be supplied in order to retrieve MANE Transcript data.
+        Liftovers genomic coordinates to GRCh38
+
+        :param str chromosome: Chromosome. Must either give chromosome number
+            (i.e. `1`) or accession (i.e. `NC_000001.11`).
+        :param int start: Start genomic position
+        :param int end: End genomic position
+        :param str strand: Strand. Must be either `-1` or `1`.
+        :param str transcript: The transcript to use. If this is not given,
+            we will try the following transcripts: MANE Select, MANE Clinical
+            Plus, Longest Remaining Compatible Transcript
+        :param str gene: Gene symbol
+        :param str residue_mode: Default is `resiude` (1-based).
+            Must be either `residue` or `inter-residue` (0-based).
+        :return: Genomic data (inter-residue coordinates)
+        """
+        params = {key: None for key in GenomicData.__dict__["__fields__"].keys()}  # noqa: E501
```

Comment:
`GenomicData.__fields__.keys()` is a little shorter and I think it does the same thing?